src/query/v2/request_router.hpp

```diff
@@ -719,4 +742,74 @@ class RequestRouter : public RequestRouterInterface {
   io::Notifier notifier_ = {};
   // TODO(kostasrim) Add batch prefetching
 };
+
+class RequestRouterFactory {
+ protected:
+  using LocalTransport = io::Io<io::local_transport::LocalTransport>;
+  using SimulatorTransport = io::Io<io::simulator::SimulatorTransport>;
```

Comment:
This naming can be confusing. If you need to keep them, please consider renaming them (maybe `LocalTransportIo`) in order to clearly  differentiate between the `Io` object and the underlying transport.